app/services/availability_service.py

```
@@ -217,14 +217,16 @@ def _resolve_slot_price(self, room_detail, date_str: str, slot: str) -> int:
 
         return max(slot_price, 0)
 
-    def calculate_total_price(self, room_detail, date_str: str, hour_slots: List[str]) -> int:
+    def calculate_total_price(self, room_detail, date_str: str, hour_slots: List[str], available_slots: Optional[Dict[str, Union[bool, str]]] = None) -> int:
```

Ruff (0.15.2)

[error] Undefined name `Union`

(F821)